File: app/src/main/java/com/example/schuller/carscharging/di/AppComponent.java
package com.example.schuller.carscharging.di;

import com.example.schuller.carscharging.driver.AvailableStation;

import javax.inject.Singleton;

import dagger.Component;

/**
 * AppComponent
 * Created by paul on 23/05/2018.
 */

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {

    void inject(AvailableStation availableStation);

}

File: app/src/main/java/com/example/schuller/carscharging/di/AppComponent.kt
package com.example.schuller.carscharging.di


import com.example.schuller.carscharging.driver.StationActivity
import com.example.schuller.carscharging.driver.StationViewModel
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [AppModule::class])
interface AppComponent {

    fun inject(stationViewModel: StationViewModel)
    fun inject(stationActivity: StationActivity)

}